modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/utils/getFields.ts

```diff
@@ -68,21 +73,30 @@ function getValidFields({
 		return fields;
 	}
 
+	if (xFilterable.length) {
+		const parentsFilterable = xFilterable
+			.filter((item) => item.includes('/'))
+			.map((item) => item.split('/')[0]);
+
+		xFilterable = [...xFilterable, ...parentsFilterable];
+	}
```

Comment:
do we need this logic on every recursive call to `getValidFields()` ? If  not, then we can move it to `getFields()` (around line 188). Makes sense?